src/im_mad/remotes/node-probes.d/pci.rb

```diff
@@ -90,6 +91,11 @@ def get_devices(filter = nil)
     end.flatten
 end
 
+def device_attr?(device, attribute)
+    addr = "0000:#{device[:bus]}:#{device[:slot]}.#{device[:function]}"
+    !`ls -l /sys/bus/pci/devices/#{addr}/ | grep #{attribute}`.empty?
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Detected a dangerous subshell execution vulnerability where user input is directly used in a shell command without proper sanitisation. This could allow an attacker to inject and execute arbitrary system commands, potentially leading to system compromise or data breach. To remediate, avoid using shell commands with user input when possible, or implement strict input validation and use shell escaping functions like Shellwords.escape(). If shell execution is necessary, consider using a allow list of allowed commands and arguments. Link to more info : https://guides.rubyonrails.org/security.html